tooling/nargo_cli/src/cli/fuzz_cmd.rs

```diff
@@ -0,0 +1,489 @@
+                        &file_manager,
+                        &parsed_files,
+                        package,
+                        &pattern,
+                        &args.compile_options,
+                    );
+                    match harnesses {
+                        Ok(harness_names) => Ok((package.name.clone(), harness_names)),
+                        Err(cli_error) => Err(cli_error),
+                    }
+                })
+            })
+            .collect::<Result<_, _>>()?;
+        let mut found_harness = false;
+        for (crate_name, discovered_harnesses) in all_harnesses_by_package.iter() {
+            if !discovered_harnesses.is_empty() {
+                println!("Package {crate_name} contains fuzzing harnesses:");
+                for harness in discovered_harnesses.iter() {
+                    println!("\t{harness}");
+                }
+                found_harness = true;
+            }
+        }
+        if !found_harness {
+            println!("No fuzzing harnesses found");
+        }
+        return Ok(());
+    }
+
+    let fuzz_folder_config = FuzzFolderConfig {
+        corpus_dir: args.corpus_dir,
+        minimized_corpus_dir: args.minimized_corpus_dir,
+        fuzzing_failure_dir: args.fuzzing_failure_dir,
+    };
+    let fuzz_execution_config =
+        FuzzExecutionConfig { timeout: args.timeout.unwrap_or(0), num_threads: args.num_threads };
+
+    let fuzzing_reports: Vec<Vec<(String, FuzzingRunStatus)>> = workspace
+        .into_iter()
+        .map(|package| {
+            run_fuzzers::<Bn254BlackBoxSolver>(
+                &file_manager,
+                &parsed_files,
+                package,
+                &pattern,
+                args.show_output,
+                args.oracle_resolver.as_deref(),
+                Some(workspace.root_dir.clone()),
+                Some(package.name.to_string()),
+                &args.compile_options,
+                &fuzz_folder_config,
+                &fuzz_execution_config,
+            )
+            .unwrap_or_else(|_| Vec::new())
+        })
+        .collect();
+
+    let fuzzing_report: Vec<(String, FuzzingRunStatus)> =
+        fuzzing_reports.into_iter().flatten().collect();
+
+    if fuzzing_report.is_empty() {
+        match &pattern {
+            FunctionNameMatch::Exact(pattern) => {
+                let single_pattern = pattern[0].clone();
+                return Err(CliError::Generic(format!(
+                    "Found 0 fuzzing_harnesses matching input '{single_pattern}'.",
+                )));
+            }
+            FunctionNameMatch::Contains(pattern) => {
+                let single_pattern = pattern[0].clone();
+                return Err(CliError::Generic(format!(
+                    "Found 0 fuzzing_harnesses containing '{single_pattern}'.",
+                )));
+            }
+            // If we are running all tests in a crate, having none is not an error
+            FunctionNameMatch::Anything => {}
```

Comment:
Should it be an error for fuzzing? Maybe someone used `#[test]` instead of `#[fuzz]`, ran the fuzzer, it produced no errors so they would think there's no issue.